[IV-23-24] Objetivo 5: Creación de un contenedor para pruebas

marinajcs opened the pull request:
* [x] ¿Se han establecido criterios previos de elección de la imagen base, así
como los criterios de búsqueda que han conducido a las imágenes que se están
considerando?
* [x] ¿Se ha tenido en cuenta que la imagen es para testear, y que por tanto
tiene una función específica, tal como testear cómo va nuestra aplicación en la
última versión publicada del lenguaJe?
* [x] ¿Se ha tenido en cuenta que la versión de la imagen base sea la última?
* [x] ¿Se han tenido en cuenta criterios reales para la elección, y no criterios
irrelevantes como cuanto tiempo se tarda en construirla? (Se va a construir una
vez, se va a usar cientos o miles de veces)
* [x] ¿Se han seguido las buenas prácticas en las órdenes que construyen
   la imagen? ¿Especialmente las relativas a usuarios no privilegiados y demás?
* [x] ¿La imagen incluye sólo y exclusivamente la infraestructura
   necesaria para pasar los tests? Por ejemplo, si se ha elegido la imagen
   "oficial" (lo que hay que hacer sólo en casos muy justificados) ¿se ha
   comprobado que lo que se instale sea sólo y exclusivamente lo necesario para
   nuestro proyecto?
* [x] ¿Se han documentado y enlazado los commits a las imágenes que se
   han testeado?
* [x] ¿Está establecido correctamente como `WORKDIR` el que se va a usar en los
   tests que se van a lanzar?
* [x] ¿Se usa algún directorio adecuado para copiar contenidos del repositorio
en vez del directorio de trabajo?
* [x] ¿Se han establecido claramente los criterios de búsqueda y de aceptación de
   la imagen base?
* [x] ¿Tienes claro cuales son las condiciones en las que tiene que actualizarse
la imagen en DockerHub y has configurado el workflow de acuerdo con ello? En
particular, ¿sabes qué condiciones tienen que darse para que se actualice y qué
condiciones *no* deben darse?
* [x] Si se usan GitHub actions para su publicación, ¿se han considerado
cuidadosamene los criterios (ramas, paths) para que se lancen?
* [x] ¿Se ha comprobado que el ENTRYPOINT del contenedor no escriba nada en el
directorio que monta, ya que no tiene derecho de escritura?
* [x] ¿Se ha abierto un shell en el contenedor una vez construida la imagen y
comprobado que los ficheros que se necesitan están todos incluidos, y que no hay
ficheros que no se necesiten?

marinajcs committed:
Elección de imagen base, closes #32

Se documenta la elección de imagen base

marinajcs committed:
Cambios en la elección de imagen, closes #32

Se añade toda la documentación relevante a la evaluación de criterios a la hora de seleccionar una imagen base.

marinajcs committed:
Fichero Dockerfile, fixes #32

Fichero de configuración para la construcción de la imagen.

marinajcs committed:
Creado workflow docker-image, fixes #32

Creación de un archivo de configuración YAML especial que define las construcción y actualización de la imagen en Docker Hub, que se deberá realizar automáticamente en cada confirmación (push) a los paths especificados.

marinajcs committed:
Incluidas órdenes docker, fixes #32

Se completa la documentación del proyecto en README, añadiendo toda la información relevante para la elección, configuración y ejecución del docker.

marinajcs committed:
Pequeñas correcciones en eleccion-imagen, fixes #32

Correcciones sintácticas y de formato Markdown.

marinajcs committed:
Justifica elección de bookworm, fixes #32

Se añade una justificación de por qué se ha considerado bookworm-slim en concreto, y no otros "tags slim".

marinajcs committed:
Corregido rendimiento imagen, fixes #32

Rendimiento de un contenedor lo determina el tiempo de ejecución de tests, no el tiempo de build.

marinajcs committed:
Corregido Dockerfile, fixes #32

Se usa el usuario por defecto Node, se le conceden los permisos necesarios a los ficheros y se evita hacer nada en el entrypoint /app/test

marinajcs committed:
Actualizadas versiones docker-image.yml, fixes #32

marinajcs committed:
Prueba workflow docker, fixes #32

Se han configurado los secretos en el proyecto y se espera a que realice el action correctamente

marinajcs committed:
Creado DOCKER_USER, fixes #32

Indica el nick de Docker Hub dado que no coincide con el de Github

marinajcs committed:
Definición de PATH en node_modules, closes #34

A pesar de detectar el package.json, no encuentra node_modules. Se establece la ruta del .bin.

migueruiz commented on Dockerfile:
```diff
@@ -0,0 +1,25 @@
+FROM node:bookworm-slim
+
+LABEL maintainer="marinacarranza@correo.ugr.es"
```
Deberías de indicar también la versión. Puedes mirar en mi repositorio si lo necesitas. 😆

marinajcs replied:
Gracias por la sugerencia :smile:
Añadido en: https://github.com/marinajcs/asignacionTareas/pull/33/commits/da87101c49a7365c82fdf060ef484e4de74f3a39

migueruiz reviewed (approved):
Hola, seré tu revisor! Veo todo tu documento correctamente, por ponerle alguna pega, lo que te he dejado abajo. Pero en principio, lo veo todo correctamente! 😄

marinajcs committed:
Incluida versión en Dockerfile, fixes #32

Se añade la versión en el Dockerfile para mejorar la claridad, documentación y gestión de cambios

marinajcs commented:
@JJ Listo para revisión

JJ commented on Dockerfile:
```diff
@@ -0,0 +1,26 @@
+WORKDIR /app
+
+COPY --chown=node:node package.json pnpm-lock.yaml tsconfig.json ./
+COPY --chown=node:node lib lib
```
No puedes copiar los contenidos del repositorio; se supone que esos contenidos tienen que estar presente sólo cuando se monte el repositio.

marinajcs replied:
Resuelto en: https://github.com/marinajcs/asignacionTareas/pull/33/commits/b6723ac855ffa2d2f30fc51c662ef7876bed2126

JJ reviewed (changes requested):
Por favor, consulta también los [errores frecuentes](https://github.com/JJ/IV-/blob/master/errores/errores-objetivo-5.md), sobre todo el segundo y el tercerlo.

marinajcs committed:
No se copian ficheros de repo en imagen, fixes #32

No se copian los contenidos del repositorio, tienen que estar presente solo cuando se monte.

JJ commented:
¿Está listo para revisión?

marinajcs commented:
> ¿Está listo para revisión?

Tratando de resolver el problema https://github.com/marinajcs/asignacionTareas/pull/33#discussion_r1429245078, ha surgido un nuevo issue (https://github.com/marinajcs/asignacionTareas/issues/35) que estuve tratando de solucionar, pero no he llegado a ninguna conclusión aún, así que había pensando en consultárselo en la clase de mañana (o el viernes).

marinajcs committed:
Probando node-red, #32

marinajcs committed:
Instalado pnpm con corepack, fixes #35

Se descargar pnpm con corepack

JJ reviewed (approved):
👍

JJ commented:
Y muchas gracias a @migueruiz por la revisión.
